docs/source/noise_models.rst

```diff
@@ -0,0 +1,523 @@
+
+Simulating gate noise on the Rigetti Quantum Virtual Machine
+============================================================
+
+
+Quantum gate errors
+-------------------
+
+For a quantum gate given by its unitary operator :math:`U`, a "quantum
+gate error" describes the scenario in which the actually induces
+transformation deviates from :math:`\ket{\psi} \mapsto U\ket{\psi}`.
+There are two basic types of quantum gate errors:
+
+1. **coherent errors** are those that preserve the purity of the input
```

Comment:
Would be good to define purity somewhere simple.  E.g. "Errors in quantum computing can introduce uncertainty in what the underlying state is.  When this happens we sometimes need to consider not only wavefunctions but also probabilistic sums of wavefunctions when we are uncertain as to which one we have.  For example, if we think that an `X` gate was accidentally applied to a qubit with a 50-50 chance then we would say that there is a 50% chance we have the |0> state and a 50% chance that we have a |1> state.  This is called an "impure" state in that it isn't just a wavefunction (which is pure) but instead a distribution over wavefunctions.  We describe this with something called a density matrix [insert section with example definition of density matrix].

Note that classical uncertainty in the wavefunction is markedly different from superpositions. We can represent superpositions using wavefunctions, but use density matrices to describe distributions over wavefunctions. You can read more about density matrices here [cite to Mike and Ike or online lecture notes somewhere]."

I'm writing quickly @ntezak but I think you get the idea.